crates/wasmtime/src/func.rs

```diff
@@ -553,14 +567,21 @@ impl Func {
         }
 
         // Call the trampoline.
-        invoke_wasm_and_catch_traps(self.export.vmctx, &self.instance.store, || unsafe {
-            (self.trampoline)(
-                self.export.vmctx,
-                ptr::null_mut(),
-                self.export.address,
-                values_vec.as_mut_ptr(),
-            )
-        })?;
+        unsafe {
+            debug_assert!(!self.export.anyfunc.as_ref().func_ptr.is_null(), "d");
```

Comment:
This expression happens a lot in this block, so perhaps this could do this at the top?

```rust
let anyfunc = self.export.anyfunc.as_ref();
// ...
```